### PR TITLE
feat: do not break math

### LINKED ISSUE #60: skip $$ $$ and $ $ in qmd for translation
It would be great if one could skip equation content from translation bt deepL in quarto qmd as it produces unexpected results.

### LINKED ISSUE #64: crossreferences can be translated or not
Sometimes cross ref are translated, sometimes not.

```qmd
---
title: "crossref"
description: "comment protéger les références croisées dans quarto et babeldown ?"
---

Imaginons une référence croisée à un chapitre qui suit. 

La référence n'est pas protégée quand elle est mise @sec-chapitre ou sous la forme [@sec-chapitre],
mais @sec-chapitre2 est protégée.

## Le chapitre suivant {#sec-chapitre}

## un autre chapitre {#sec-chapitre2}
```

is translated from french to english as (note the @sec-chapter):

```qmd
---
title: crossref
description: how do you protect cross-references in quarto and babeldown?
---

Let's imagine a cross-reference to a chapter that follows.

The reference is not protected when it is put in @sec-chapter or in the form [@sec-chapter],
but @sec-chapitre2 is protected.

## The next chapter {#sec-chapitre}

## another chapter {#sec-chapitre2}
``` 

A way is to add some number to all cross-refs, but I think there is another solution ?

### LINKED ISSUE #65: equations and footnote don't play nicely
When either a footnote is on the same line as an equation or when an equation is in the footnote some spaces disapear

```qmd
---
title: "EqinFN"
description: "Une équation et une note de bas de page et c'est le bazar !"
---
sans note de bas de page, c'est ok $a$ est bien conservé dans sa beauté initiale

mais $a$ une équation avec une nbp ensuite [^1]

[^1]: une note de bas de page

mais quand il l'équation est dans la note de bas de page [^2]

[^2]: une équation $a$ dans une note de bas de page
```
gets translated to (please note the missing space after `$a$` the first time and before `$a$` the second time and that the semicolon is also missing, resulting in a dead footnote):

```qmd
---
title: EqinFN
description: An equation and a footnote and it's a mess!
---

without a footnote, it's OK $a$ is well preserved in its original beauty

but $a$an equation with a nbp then [^1]

[^1] a footnote

but when the equation is in the footnote [^2]

[^2] an equation$a$ in a footnote
```

### PR DESCRIPTION
**Pull request by maelle**
Fix #60
Fix #61
Fix #64
Fix #65

@xtimbeau could you please have a look?

### CODE REVIEW

**Comment by maelle** [comment]
I need to try the case with a one-character equation though.

File: R/translate.R [hunk 1]
```diff
@@ -191,7 +194,7 @@ translate_part <- function(xml,
       non_splitting_tags = "text,softbreak",
       formality = formality,
       glossary_id = glossary_id,
-      ignore_tags = "code,code_block,curly,notranslate"
+      ignore_tags = "code,code_block,curly,math,notranslate"
```
Comment (maelle) [inline R/translate.R:197]: because of this I had to re-record all fixtures :upside_down_face:

File: tests/testthat/_snaps/translate.md [hunk 2]
```diff
@@ -25,3 +25,17 @@
       Error in `deepl_translate()`:
       ! `formality` must be one of "default", "more", "less", "prefer_more", or "prefer_less", not "non-existing-formality".
 
+# deepl_translate() handles equations well
+
+    Code
+      math_lines[4]
+    Output
+      [1] "E = m \\times c^2"
```
Comment (maelle) [inline tests/testthat/_snaps/translate.md:33]: this actually looks good in the output file, but that's how the backslash has to be recorded here.

**Comment by xtimbeau** [comment]
Hi @maelle, `protect_math` is working, but there are still 3 problems (that I know).
1. One char equations are throwing an error #
2. multiple `_`in latex are (wrongly) transformed to <emph> tag
3. there is still the problem of footnote equation #65  

In #61 I proposed a way to deal with 1 and 2 (not very satisfying, I agree) by adding chararecters (`xXx` sequence) to one char equation (which deals with the bug) and by transforming "_" to "°°" before translation. Both replacements are reversed after the translation, so everything is invisible to the user. If protect_math was able to deal with those cases, than those 2 workarounds would be useless.

**Comment by maelle** [comment]
@xtimbeau thanks!

> One char equations are throwing an error #

Have you installed tinkr from my branch? Running `pak::pak()` should do that automatically. 

> there is still the problem of footnote equation

See my latest commit, I was working on that.

> multiple _in latex are (wrongly) transformed to tag

Any example I could use?

File: inst/example-equations.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Some text.
+
+$$
+E = m \times c^2
+$$
+
+How nice is this equation? Note that $\alpha=1$ and $b$ is not defined.
+
+$i_t = j_t$
```
Comment (maelle) [inline inst/example-equations.md:9]: @xtimbeau I had added this.

Comment (maelle) [reply inst/example-equations.md:9]: Great! I'll merge this if the checks pass. Then you can report new bugs in the coming weeks :smile_cat:

**Pull request merged**